FLT/DivisionAlgebra/Finiteness.lean

```diff
@@ -172,8 +173,75 @@ lemma antidiag_mem_C {β : D_𝔸ˣ} (hβ : β ∈ ringHaarChar_ker D_𝔸) :
 
 end Aux
 
-lemma compact_quotient : CompactSpace (ringHaarChar_ker D_𝔸 ⧸
-  (MonoidHom.range (incl K D)).comap (ringHaarChar_ker D_𝔸).subtype) := sorry
+/-- The inclusion of `ringHaarChar_ker D_𝔸` into the product space `D_𝔸 × D_𝔸ᵐᵒᵖ`. -/
+def incl₂ : ringHaarChar_ker D_𝔸 → Prod D_𝔸 D_𝔸ᵐᵒᵖ :=
+  fun u => Units.embedProduct D_𝔸 (Subgroup.subtype (ringHaarChar_ker D_𝔸) u)
+
+/-- An auxillary set used in the proof of compact_quotient'. -/
+def M : Set (ringHaarChar_ker D_𝔸) := Set.preimage (incl₂ K D)
+  (Set.image (fun p => (p.1, MulOpposite.op p.2)) (Aux.C K D))
+
+/-- The map from `ringHaarChar_ker D_𝔸` to the quotient `Dˣ \ ringHaarChar_ker D_𝔸`. -/
+abbrev toQuot (a : ringHaarChar_ker D_𝔸) : (_root_.Quotient (QuotientGroup.rightRel
+    ((MonoidHom.range (incl K D)).comap (ringHaarChar_ker D_𝔸).subtype))) :=
+  (Quotient.mk (QuotientGroup.rightRel ((MonoidHom.range (incl K D)).comap
+  (ringHaarChar_ker D_𝔸).subtype)) a)
+
+lemma toQuot_cont : Continuous (toQuot K D) := by exact { isOpen_preimage := fun s a ↦ a }
+
+lemma toQuot_surjective : (toQuot K D) '' (M K D) = Set.univ := by
+  rw [Set.eq_univ_iff_forall]
+  rintro ⟨a, ha⟩
+  obtain ⟨c, hc, ν, hν, rfl, h31⟩ := Aux.antidiag_mem_C K D ha
+  simp only [toQuot, Subgroup.comap_subtype, Set.mem_image, Subtype.exists]
+  refine ⟨ν, hν, ?_, ?_ ⟩
+  · simp only [M, Set.mem_preimage, Set.mem_image, Prod.exists]
+    exact ⟨ν, Units.val (ν⁻¹), h31, rfl⟩
+  · have : Quot.mk ⇑(QuotientGroup.rightRel ((incl K D).range.subgroupOf
+        (ringHaarChar_ker (D ⊗[K] AdeleRing (𝓞 K) K)))) ⟨c * ν, ha⟩ =
+        Quot.mk ⇑(QuotientGroup.rightRel ((incl K D).range.subgroupOf
+        (ringHaarChar_ker (D ⊗[K] AdeleRing (𝓞 K) K))))
+        ⟨ν, hν⟩ := by
+      refine Quot.sound ?_
+      rw [@QuotientGroup.rightRel_apply]
+      refine Subgroup.mem_subgroupOf.mpr ?_
+      simp only [@Subgroup.coe_mul, InvMemClass.coe_inv, mul_inv_rev, mul_inv_cancel_left,
+        inv_mem_iff, MonoidHom.mem_range]
+      exact hc
+    rw [this]
+    rfl
+
+-- elsewhere
+local instance : T2Space (D ⊗[K] AdeleRing (𝓞 K) K) := by
+
+  sorry
```

Comment:
If you (a) wait for #669 to be merged (which should happen later today) and then (b) merge main, then this sorry should just be `IsModuleTopology.t2space (AdeleRing (𝓞 K) K)`.